Mathlib/LinearAlgebra/Matrix/SchurComplement.lean

```diff
@@ -445,88 +444,4 @@ end Det
 
 end CommRing
 
-/-! ### Lemmas about `ℝ` and `ℂ` and other `StarOrderedRing`s -/
-
-
-section StarOrderedRing
-
-variable {𝕜 : Type*} [CommRing 𝕜] [StarRing 𝕜]
-
-/-- Notation for `Sum.elim`, scoped within the `Matrix` namespace. -/
-scoped infixl:65 " ⊕ᵥ " => Sum.elim
-
-theorem schur_complement_eq₁₁ [Fintype m] [DecidableEq m] [Fintype n] {A : Matrix m m 𝕜}
-    (B : Matrix m n 𝕜) (D : Matrix n n 𝕜) (x : m → 𝕜) (y : n → 𝕜) [Invertible A]
-    (hA : A.IsHermitian) :
-    (star (x ⊕ᵥ y)) ᵥ* (fromBlocks A B Bᴴ D) ⬝ᵥ (x ⊕ᵥ y) =
-      (star (x + (A⁻¹ * B) *ᵥ y)) ᵥ* A ⬝ᵥ (x + (A⁻¹ * B) *ᵥ y) +
-        (star y) ᵥ* (D - Bᴴ * A⁻¹ * B) ⬝ᵥ y := by
-  simp [Function.star_sumElim, vecMul_fromBlocks, add_vecMul,
-    dotProduct_mulVec, vecMul_sub, Matrix.mul_assoc, hA.eq,
-    conjTranspose_nonsing_inv, star_mulVec]
-  abel
-
-theorem schur_complement_eq₂₂ [Fintype m] [Fintype n] [DecidableEq n] (A : Matrix m m 𝕜)
-    (B : Matrix m n 𝕜) {D : Matrix n n 𝕜} (x : m → 𝕜) (y : n → 𝕜) [Invertible D]
-    (hD : D.IsHermitian) :
-    (star (x ⊕ᵥ y)) ᵥ* (fromBlocks A B Bᴴ D) ⬝ᵥ (x ⊕ᵥ y) =
-      (star ((D⁻¹ * Bᴴ) *ᵥ x + y)) ᵥ* D ⬝ᵥ ((D⁻¹ * Bᴴ) *ᵥ x + y) +
-        (star x) ᵥ* (A - B * D⁻¹ * Bᴴ) ⬝ᵥ x := by
-  simp [Function.star_sumElim, vecMul_fromBlocks, add_vecMul,
-    dotProduct_mulVec, vecMul_sub, Matrix.mul_assoc, hD.eq,
-    conjTranspose_nonsing_inv, star_mulVec]
-  abel
-
-theorem IsHermitian.fromBlocks₁₁ [Fintype m] [DecidableEq m] {A : Matrix m m 𝕜} (B : Matrix m n 𝕜)
-    (D : Matrix n n 𝕜) (hA : A.IsHermitian) :
-    (Matrix.fromBlocks A B Bᴴ D).IsHermitian ↔ (D - Bᴴ * A⁻¹ * B).IsHermitian := by
-  have hBAB : (Bᴴ * A⁻¹ * B).IsHermitian := by
-    apply isHermitian_conjTranspose_mul_mul
-    apply hA.inv
-  rw [isHermitian_fromBlocks_iff]
-  constructor
-  · intro h
-    apply IsHermitian.sub h.2.2.2 hBAB
-  · intro h
-    refine ⟨hA, rfl, conjTranspose_conjTranspose B, ?_⟩
-    rw [← sub_add_cancel D]
-    apply IsHermitian.add h hBAB
-
-theorem IsHermitian.fromBlocks₂₂ [Fintype n] [DecidableEq n] (A : Matrix m m 𝕜) (B : Matrix m n 𝕜)
-    {D : Matrix n n 𝕜} (hD : D.IsHermitian) :
-    (Matrix.fromBlocks A B Bᴴ D).IsHermitian ↔ (A - B * D⁻¹ * Bᴴ).IsHermitian := by
-  rw [← isHermitian_submatrix_equiv (Equiv.sumComm n m), Equiv.sumComm_apply,
-    fromBlocks_submatrix_sum_swap_sum_swap]
-  convert IsHermitian.fromBlocks₁₁ _ _ hD <;> simp
-
-variable [PartialOrder 𝕜] [StarOrderedRing 𝕜]
-
-theorem PosSemidef.fromBlocks₁₁ [Fintype m] [DecidableEq m] [Fintype n] {A : Matrix m m 𝕜}
```

Comment:
This is mentioned in the module docstring; can you move that docstring with the lemma?